utils/src/main/scala/com/ibm/sparktc/sparkbench/utils/GeneralFunctions.scala

```diff
@@ -19,47 +19,30 @@ package com.ibm.sparktc.sparkbench.utils
 
 import java.io.StringWriter
 
-import scala.util.{Failure, Random, Success, Try}
-
+import scala.sys.process._
+import scala.util.{Failure, Success, Try}
 
 object GeneralFunctions {
+  private val log = org.slf4j.LoggerFactory.getLogger(getClass)
 
-  val any2Long = (a: Any) => {
-    a.asInstanceOf[Number].longValue()
+  val any2Long: (Any) => Long = {
+    case x: Number => x.longValue
+    case x => x.toString.toLong
   }
 
-  def getOrDefault[A](
-                       map: Map[String, Any],
-                       name: String, default: A,
-                       func: (Any) => A = {(any: Any) => any.asInstanceOf[A]}
-                     ): A = {
-
-    val any = map.get(name) match {
-      case None => return default
-      case Some(a) => a
-    }
+  private def defaultFn[A](any: Any): A = any.asInstanceOf[A]
 
-    Try(func(any)) match {
-      case Success(b) => b
-      case Failure(_) => default
-    }
+  private[utils] def tryWithDefault[A](a: Any, default: A, func: Any => A): A = Try(func(a)) match {
+    case Success(b) => b
+    case Failure(_) => default
   }
 
-  def getOrDefaultOpt[A](
-                       opt: Option[Any],
-                       default: A,
-                       func: (Any) => A = {(any: Any) => any.asInstanceOf[A]}
-                     ): A = {
-
-    val any = opt match {
-      case None => return default
-      case Some(a) => a
-    }
+  def getOrDefaultOpt[A](opt: Option[Any], default: A, func: Any => A = defaultFn[A](_: Any)): A = {
+    opt.foldLeft(default) { case (d, a) => tryWithDefault(a, d, func) }
```

Comment:
i think opt.map(func).getOrElse(default) is clearer

Reply:
good call!